centreon/www/include/configuration/configObject/hostgroup/DB-Func.php

```diff
@@ -1178,3 +990,155 @@ function getPollersForConfigChangeFlagFromHostgroupId(int $hostgroupId): array
     $hostIds = findHostsForConfigChangeFlagFromHostGroupIds([$hostgroupId]);
     return findPollersForConfigChangeFlagFromHostIds($hostIds);
 }
+
+// ---------- API CALLs ----------
+
+/**
+ * Create a new host group from formData.
+ *
+ * @return int|null
+ */
+function insertHostGroup(): int|false
+{
+    global $centreon, $form, $isCloudPlatform, $basePath;
+
+    try {
+        /** @var array<string,int|string|null> $formData */
+        $formData = $form->getSubmitValues();
+        if (null === ($hostGroupId = insertHostGroupByApi($formData, $isCloudPlatform, $basePath))) {
+            throw new Exception('New hostgroup ID invalid');
+        }
+
+        updateHostGroupHosts($hostGroupId, $formData);
+        updateHostgroupAcl($hostGroupId, $isCloudPlatform, $formData);
+        signalConfigurationChange('hostgroup', $hostGroupId);
+        $centreon->user->access->updateACL();
+
+        return $hostGroupId;
+    } catch (JsonException $ex) {
+        CentreonLog::create()->error(CentreonLog::TYPE_BUSINESS_LOG, 'Error during host group creation',
+        [
+            'hostGroupId' => $hostGroupId ?? null,
+            'exception' => ['message' => $ex->getMessage(), 'trace' => $ex->getTraceAsString()],
+        ]);
+        echo "<div class='msg' align='center'>" . _('Error during creation (json encoding). See logs for more detail') . '</div>';
```

Comment:
Technical errors do not have to be displayed to the user.
Don't forget that users don't necessarily have access to Centreon log files.